Mathlib/Data/Set/Function.lean

```diff
@@ -624,6 +624,45 @@ lemma bijOn_id (s : Set α) : BijOn id s s := ⟨s.mapsTo_id, s.injOn_id, s.surj
 theorem BijOn.comp (hg : BijOn g t p) (hf : BijOn f s t) : BijOn (g ∘ f) s p :=
   BijOn.mk (hg.mapsTo.comp hf.mapsTo) (hg.injOn.comp hf.injOn hf.mapsTo) (hg.surjOn.comp hf.surjOn)
 
+/-- If `f : α → β` and `g : β → γ` and if `f` is injective on `s`, then `f ∘ g` is a bijection
+on `s` iff  `g` is a bijection on `f '' s`. -/
+theorem bijOn_comp_iff_bijOn_image {α β γ : Type*} {f : α → β} {g : β → γ} {s : Set α}
+    (hf : InjOn f s) {t : Set γ} : BijOn (g ∘ f) s t ↔ BijOn g (f '' s) t := by
```

Comment:
Could you please add missing `Iff` lemmas about `SurjOn` and `InjOn` so that this becomes `simp only [BijOn, relevant_lemmas, hf]`?

Reply:
Very nice idea! Do you think `injOn_comp_iff`, `surjOn_comp_iff` and `bijOn_comp_iff` are too brief? They feel like the most natural lemmas because you are "moving f from the function to the set".